embedded/store/immustore.go

```diff
@@ -2566,35 +2566,31 @@ func (s *ImmuStore) ReadValue(entry *TxEntry) ([]byte, error) {
 }
 
 func (s *ImmuStore) readValueAt(b []byte, off int64, hvalue [sha256.Size]byte) (n int, err error) {
-	mustReadFromVLog := true
-
 	if s.vLogCache != nil {
 		val, err := s.vLogCache.Get(off)
 		if err == nil {
 			// the requested value was found in the value cache
 			copy(b, val.([]byte))
-			mustReadFromVLog = false
+			return len(b), nil
```

Comment:
the flag was intentional so to do the digest comparison, otherwise the validation against the expected hvalue won't be done

Reply:
can we do the validation only when len(b)>0?

Reply:
validation should be done in all the cases

Reply:
added validation for cached value